tensorflow/contrib/tensorrt/convert/convert_nodes.cc

```diff
@@ -2633,6 +2856,54 @@ tensorflow::Status ConvertTopK(Converter& ctx,
 }
 #endif
 
+void ApplyQuantizationRanges(Converter& converter, bool use_calibration) {
+  std::vector<nvinfer1::ITensor*> tensors_missing_ranges;
+  // Infer quantization ranges backward through reshapes, maxpool, etc
+  for (auto it : converter.quantization_infer_backwards_) {
```

Comment:
IIUC, the algorithm here is, use quantization_infer_backwards_[output_tensor] to populate the range information of the input tensor, but only do this once. What if there are multiple hops, e.g. `tensor1->reshape1->tensor2->identity1->tensor3` and we only know the range of `tensor3`? And what if `tensor1` is an input of a different `reshape2` to `tensor4`?

It looks like the right way would be:
```
while (not more range information are added) {
  do infer backwards
  do infer forwards
}
```
what do you think? Thanks.

Reply:
Hi Lambda, thanks for all of the comments!

Yes that is the correct algorithm. My code assumed a max of one hop which was okay since we only inferred ranges for tensors that we create internally within TF-TRT  (the user could always just provide a quantization node for everything else).

But I agree that we should be smarter about inferring to make this feature easier to use, so I will implement the multi-hop algorithm.